core/domain/summary_services_test.py

```diff
@@ -196,6 +196,97 @@ def test_get_displayable_exp_summary_dicts_matching_ids(self):
         self.assertDictContainsSubset(expected_summary,
                                       displayable_summaries[0])
 
+    def test_get_displayable_exp_summary_dicts(self):
+        exp_summary_input = exp_services.get_exploration_summaries_matching_ids(
+            [self.EXP_ID_2])
+        exp_summary_input.insert(0, None)
+        exploration_summary = (
+            summary_services.get_displayable_exp_summary_dicts(
+                exp_summary_input))
+        expected_summary = {
+            'status': u'public',
+            'thumbnail_bg_color': '#05a69a',
+            'community_owned': False,
+            'tags': [],
+            'thumbnail_icon_url': '/images/gallery/thumbnails/Lightbulb.svg',
+            'language_code': feconf.DEFAULT_LANGUAGE_CODE,
+            'human_readable_contributors_summary': {
+                self.ALBERT_NAME: {
+                    'num_commits': 2,
+                    'profile_picture_data_url': None
+                }
+            },
+            'id': self.EXP_ID_2,
+            'category': u'A category',
+            'ratings': feconf.get_empty_ratings(),
+            'title': u'Exploration 2 Albert title',
+            'num_views': 0,
+            'objective': u'An objective'
+        }
+        self.assertDictContainsSubset(expected_summary,
+                                      exploration_summary[0])
+
+
+class ExplorationCategories(
+        exp_services_test.ExplorationServicesUnitTests):
+    """Test functions for getting exploration categories summary dicts."""
+
+    def setUp(self):
+        """Populate the database of explorations and their summaries.
+
+        The sequence of events is:
+        - (1) Admin logs in.
+        - (2) Admin access admin page.
+        - (3) Admin reloads exploration with id =2.
+        - (4) Admin logs out.
+        """
+
+        super(ExplorationCategories, self).setUp()
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.testapp.get('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.post_json('/adminhandler', {
+            'action': 'reload_exploration',
+            'exploration_id': '2'
+        }, csrf_token)
+        self.logout()
+
+    def test_get_gallery_category_groupings(self):
+        """Exploration with id property of 2 is an exploration with
```

Comment:
The exploration with id '2' is an exploration in the Mathematics category. The call to...